AbletonMCP_Remote_Script/handlers/midi.py

```
@@ -0,0 +1,219 @@
+        if ctrl:
+            ctrl.log_message("Error transposing clip: " + str(e))
+        raise
+
+
+def duplicate_clip(
+    song, source_track, source_clip, dest_track, dest_clip, ctrl=None
+):
+    """Duplicate a clip."""
+    try:
+        if source_track < 0 or source_track >= len(song.tracks):
+            raise IndexError("Source track index out of range")
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)